Sources/FoundationEssentials/String/String+Internals.swift

```diff
@@ -138,7 +138,13 @@ extension String {
             return try block(buffer.baseAddress!)
         }
         #else
-        try self.withCString {
+        var iter = self.utf8.makeIterator()
+        let bLeadingSlash = if iter.next() == ._slash, iter.next()?.isLetter ?? false, iter.next() == ._colon { true } else { false }
+
+        // Strip the leading `/` on a RFC8089 path (`/[drive-letter]:/...` ).  A
+        // leading slash indicates a rooted path on the drive for the current
+        // working directory.
+        return try Substring(self.utf8.dropFirst(bLeadingSlash ? 1 : 0)).withCString {
             try block($0)
         }
         #endif
```

Comment:
Do we need to apply this to the `withMutableFileSystemRepresentation` function below as well?